### PR TITLE
Add Debt Payments to Vault.report

### PR DESCRIPTION
**Pull request by fubuloubu**
Based on: #81

### CODE REVIEW

File: contracts/Vault.vy [hunk 1]
```diff
@@ -1190,50 +1190,22 @@ def expectedReturn(_strategy: address = msg.sender) -> uint256:
...
-        self.strategies[msg.sender].totalDebt -= loss
-        self.totalDebt -= loss
-        # Also, make sure we reduce our trust with the strategy by the same amount
-        self.strategies[msg.sender].debtLimit -= min(
-            loss,
-            self.strategies[msg.sender].debtLimit,
-        )
+    # Also, make sure we reduce our trust with the strategy by the same amount
+    debtLimit: uint256 = self.strategies[_strategy].debtLimit
+    self.strategies[_strategy].debtLimit -= min(loss, debtLimit)
```
Comment (Macarse) [inline contracts/Vault.vy:1204]: is this `min` necessary?

**Review by Macarse — APPROVED** [review]

File: contracts/Vault.vy (hunk 1 above)
Comment (fubuloubu) [reply contracts/Vault.vy:1204]: Just in case the strategy reports loss incorrectly, we want to protect against an underflow here

**Review by Grandthrax — COMMENTED** [review]
All looks good.

File: contracts/BaseStrategy.sol [hunk 2]
```diff
@@ -232,12 +224,19 @@ abstract contract BaseStrategy {
...
+
      */
-    function prepareReturn(uint256 _debtOutstanding) internal virtual returns (uint256 _profit, uint256 _loss);
+    function prepareReturn(uint256 _debtOutstanding)
+        internal
+        virtual
+        returns (
+            uint256 _profit,
+            uint256 _loss,
+            uint256 _debtPayment
```
[the remark this answers is not in the document]
Comment (fubuloubu) [reply contracts/BaseStrategy.sol:238]: Yes, `debtOustanding` is only a guide. Payments should happen with minimal slippage

**Review by andy8052 — APPROVED** [review]
This looks solid to me

File: contracts/BaseStrategy.sol [hunk 3]
```diff
@@ -228,16 +220,27 @@ abstract contract BaseStrategy {
...
-     * Perform any strategy unwinding or other calls necessary to capture
-     * the "free return" this strategy has generated since the last time it's
-     * core position(s) were adusted. Examples include unwrapping extra rewards.
-     * This call is only used during "normal operation" of a Strategy, and should
-     * be optimized to minimize losses as much as possible. It is okay to report
-     * "no returns", however this will affect the credit limit extended to the
-     * strategy and reduce it's overall position if lower than expected returns
-     * are sustained for long periods of time.
+     * Perform any strategy unwinding or other calls necessary to capture the "free return"
+     * this strategy has generated since the last time it's core position(s) were adusted.
```
Comment (dmolina79) [inline contracts/BaseStrategy.sol:224]: minor typo `adjusted.`

**Review by dmolina79 — APPROVED** [review]
LGTM! only minor typo

**Pull request merged**